src/scm/bitbucket.js

```diff
@@ -1,4 +1,295 @@
 "use strict";
 
 class Bitbucket {
-}
\ No newline at end of file
+  constructor(baseUrl, user, token) {
+    this.baseUrl = baseUrl;
+    this.user = user;
+    this.token = token;
+    this.accessToken = null;
+    this.namespaces = [user];
+  }
+
+  get name() {
+    return 'bitbucket';
+  }
+
+  get canUseGist() {
+    return false;
+  }  
+
+  getAccessToken() {
+    return new Promise((resolve, reject) => {
+      $.ajax({
+        url: 'https://bitbucket.org/site/oauth2/access_token',
+        headers: {
+          Authorization: `Basic RmZIVE02ZnN5NDJQQlJDRjRQOmVDZDN0TTh5TUpUeTJSMld4bTJWUzZoYWVKdnpuNzdw`
+        },
+        method: 'POST',
+        dataType: 'json',
+        contentType: 'application/x-www-form-urlencoded',
+        data: {
+          grant_type: 'refresh_token',
+          refresh_token: this.token
+        }
+      })
+      .then(resolve)
+      .fail(reject)
+    })
+    .then(response => {
+      chrome.storage.sync.set({ token: response.refresh_token });
+      this.accessToken = response.access_token;
+      return response.access_token;
+    })
+    .catch(err => {
+      showAlert('Failed to refresh access token.', LEVEL_ERROR);
+    })
+  }
+
+  commitFiles(repo, branch, files, deleteFiles, comment) {
+    return new Promise((resolve, reject) => {
+      let data = files.reduce((hash, f) => {
+        hash[f.name] = f.content;
+        return hash;
+      }, {});
+      data.message = comment;
+      if (deleteFiles && deleteFiles.length > 0) {
+        data.files = deleteFiles;
+      }
+      if (branch) {
+        data.branch = branch;
+      }
+      $.ajax({
+        url: `${this.baseUrl}/repositories/${repo}/src`,
+        headers: {
+          'Authorization': `Bearer ${this.accessToken}` 
+        },
+        contentType: 'application/x-www-form-urlencoded',
+        method: 'POST',
+        crossDomain: true,
+        traditional: true,
+        data: data,
+      })
+      .then(resolve)
+      .fail(reject);
+    });
+  }
+
+  push(code){
+    const changed = $('.diff-file:checked').toArray().map(elem => elem.value);
+    const files = changed.filter(f => code.gas[f]).map(f => {
+      return { name: f, content: code.gas[f] }
+    });
+    const deleteFiles = changed.filter(f => !code.gas[f]);
+    const comment = $('#commit-comment').val();
+
+    this.commitFiles(context.repo.fullName, context.branch, files, deleteFiles, comment)
+    .then(() => {
+      showAlert(`Successfully push to ${context.branch} of ${context.repo.fullName}`);
+    })
```

Comment:
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/leonhartX/gas-github/blob/b1257aeff34c7da1703f8a8cb7e32838448492d8/src/scm/bitbucket.js#L89"></span>‘err’ is defined but never used.</td>
    </tr>
  </tbody>
</table>